src/org/gk/slicing/SlicingEngine.java

```diff
@@ -239,6 +239,65 @@ public void slice() throws Exception {
         setStableIdReleased();
     }
     
+    /**
+     * Iterate through EntitySet instances and populate each instance's
+     * disease and compartment attributes.
+     * 
+     * <h3>Purpose</h3>
+     * Consider an EntitySet composed of all drugs in its hasMember slot, 
+     * each of which may have different set of diseases.
+     * We want to collect all diseases together into the disease slot of this EntitySet.
+     * 
+     * <h3>Recursion</h3>
+     * An EntitySet may contain another EntitySet. So this should be done recursively.
+     * The same thing should be done for the compartment slot. 
+     * 
+     * <h3>Application Set</h3>
+     * In the actual implementation, we should apply the above rule to all EntitySets,
+     * not necessary for EntitySet having drugs only.
+     * 
+     * @param GKInstance, String...
+     * @throws Exception
+     */
+    private void populateEntitySet(GKInstance instance, String... attributeNames) throws Exception {
+    	Set<GKInstance> attributes = InstanceUtilities.getContainedInstances(instance, 
+    			ReactomeJavaConstants.hasMember,
+    			ReactomeJavaConstants.hasCandidate);
+    	// If EntitySet has a "hasMember" or "hasCandidate" attribute then recursively iterate over them.
+    	if (attributes != null && attributes.size() > 0) {
+    		for (GKInstance attribute: attributes) {
+    			populateEntitySet(attribute);
+    		}
+    	}
+		getAndSetAttributes(instance, attributeNames);
+    }
+
+    /**
+     * Get and set attributes for a containing instance object.
+     * 
+     * <ul>
+     *   <li> "container" is the containing instance object (e.g. EntitySet). </li>
+     *   <li> "attributeNames" are the String names of the attribute instances (e.g. ReactomeJavaConstants.disease). </li>
+     *   <li> "value" is the actual instance object of the attribute (e.g. disease). </li>
+     * </ul>
+     * 
+     * @param container
+     * @param attributeNames
+     * @throws Exception
+     */
+    private void getAndSetAttributes(GKInstance container, String... attributeNames) throws Exception {
+		// for each attribute name
+		for (String attributeName: attributeNames) {
+			// get list of attribute instance objects.
+			List<GKInstance> attributes = container.getAttributeValuesList(attributeName);
+
+			// set slot values for each of the attribute instances.
+			for (GKInstance value : attributes) {
+				container.setAttributeValue(attributeName, value);
```

Comment:
This code is wrong. How is it possible to assign a member of an EntitySet to its compartment or disease slot? Your code is doing this exactly.

Reply:
This was a bit of bad wording on my part! My understanding is that `getAndSetAttributes ` would only set the `attributes` passed to it (`ReactomeJavaConstants.disease` and `ReactomeJavaConstants.compartment` in this case).

---

Added better wording in Fixed in [9073362](https://github.com/reactome/CuratorTool/commit/90733623c17ba46684902632223c8e4a77bd361a).